tests/upgrades/test_performance_tuning.py

```diff
@@ -122,14 +179,54 @@ def _create_custom_hiera_file(mongodb_type, tune_size="default"):
             if tune_size == "medium":
                 fd.write("\n".join(MEDIUM_TUNING_DATA))
 
+    @staticmethod
+    def _data_collection_of_set_tune_params(
+        tune_size_data, tune_params_collection_regex, tune_params_group
+    ):
+        """
+        This method is used to create the set of expected tune parameters value from the
+        available tune size and collect the actual set tune parameters value on the satellite
+        server and mapped them with their suitable keys.
+        :param list tune_size_data: List of parameters of selected tune size.
+        :param dict tune_params_collection_regex: Key, value mapped regex expression based on
+        selected tune size.
+        :param dict tune_params_group: Key, value mapped group of param's of selected size.
+
+        :return: Return the created & collected key value pair of tune parameters.
+        """
+        tune_param = dict()
+        mapped_tune_param = dict()
+        for tune_data in tune_size_data:
+            tune_data = tune_data.replace("::", "").strip()
+            if len(tune_data) > 0:
+                tune_data = tune_data.split(':')
+                if tune_data[1].strip():
+                    tune_param[tune_data[0]] = tune_data[1].strip()
```

Comment:
This will throw an error `AttributeError: 'NoneType' object has no attribute 'split'` if tune_data[1] is `None`. 

You should :
```suggestion
                if tune_data[1]:
                    tune_param[tune_data[0]] = tune_data[1].strip()
```